File: lesson6/c_c++/lesson6_3.c
/*
 * lesson6/c_c++/lesson6_3.c
 */
#include <stdio.h>

int sumDigits(int numb_n) {
    int numb_i = 0;
    while (numb_n != 0) {
        numb_i += numb_n % 10;
        numb_n /= 10;
    }
    return numb_i;
}

int main() {
    int numb = 12345;
    int numb_res = sumDigits(numb);
    printf("numb_res -> %d\n", numb_res);

    int numb_k = 54321;
    int numb_m = 12345;
    int numb_z = sumDigits(numb_k) + sumDigits(numb_m);
    printf("numb_z -> %d\n", numb_z);

    if (sumDigits(numb) % 2 == 0) {
        printf("сумма цифр четная\n");
        printf("она равна %d\n", sumDigits(numb));
    } else {
        printf("сумма цифр не четная\n");
    }
    return 0;
}

File: lesson6/python3/lesson6_3.py
#!/usr/bin/python3
#: lesson6/python3/lesson6_3.py


def sum_digits(numb_n):
    sum_numb = 0
    while numb_n != 0:
        sum_numb += numb_n % 10
        numb_n //= 10
    return sum_numb


numb = 12345
numb_sum = sum_digits(numb)
print(numb_sum)

numb_x = 54321
numb_y = 12345
numb_z = sum_digits(numb_x) + sum_digits(numb_y)
print(numb_z)

if sum_digits(numb) % 2 == 0:
    print("сумма цифр четная")
else:
    print("сумма цифр не четная")
